lib/matplotlib/tests/test_constrainedlayout.py

```diff
@@ -719,3 +725,59 @@ def test_layout_leak():
     gc.collect()
     assert not any(isinstance(obj, mpl._layoutgrid.LayoutGrid)
                    for obj in gc.get_objects())
+
+
+@image_comparison(['gridspecfromsubplotspec_wspace.png'])
+def test_gridspecfromsubplotspec_wspace_hspace():
+    fig = plt.figure(figsize=[10, 6], layout="constrained")
+    a0, a1 = fig.subplots(
+        nrows=1,
+        ncols=2,
+        gridspec_kw={"wspace": 0.3, "hspace": 0.3},
+        width_ratios=[2, 1],
+    )
+
+    for axis in [a0, a1]:
+        axis.get_xaxis().set_visible(False)
+        axis.get_yaxis().set_visible(False)
```

Comment:
The goal of constrained_layout is to deal with varying sized labels on axes.  You should not remove the tick labels.  Ideally you would also test with randomly sized labels, so in this case the inner labels could be shorter than the outer labels.